airbyte-integrations/connectors/source-freshdesk/source_freshdesk/streams.py

```diff
@@ -0,0 +1,254 @@
+
+    def path(self, **kwargs) -> str:
+        return "groups"
+
+
+class Roles(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "roles"
+
+
+class Skills(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "skills"
+
+
+class TimeEntries(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "time_entries"
+
+
+class Surveys(FreshdeskStream):
+
+    def path(self, **kwargs) -> str:
+        return "surveys"
+
+
+class Tickets(IncrementalFreshdeskStream):
+    ticket_paginate_limit = 300
+    state_checkpoint_interval = 100
+
+    def path(self, **kwargs) -> str:
+        return "tickets"
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({
+            "order_type": "asc",  # ASC order, to get the old records first
+            "order_by": "updated_at",
+        })
+        if next_page_token and "updated_since" in next_page_token:
+            params["updated_since"] = next_page_token["updated_since"]
+        return params
+    
+    def parse_link_params(self, link_query_params: Mapping[str, List[str]]) -> Mapping[str, Any]:
+        params = super().parse_link_params(link_query_params)
+        if "updated_since" in link_query_params:
+            params["updated_since"] = link_query_params["updated_since"][0]
+        return params
+    
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        """
+        This block extends Incremental stream to overcome '300 page' server error.
+        Since the Ticket endpoint has a 300 page pagination limit, after 300 pages, update the parameters with
+        query using 'updated_since' = last_record, if there is more data remaining.
+        """
+        next_page_token = super().next_page_token(response=response)
+
+        if next_page_token and int(next_page_token["page"]) > self.ticket_paginate_limit:
+            # get last_record from latest batch, pos. -1, because of ACS order of records
+            last_record_updated_at = response.json()[-1]["updated_at"]
+            last_record_updated_at = pendulum.parse(last_record_updated_at)
+            # updating request parameters with last_record state
+            next_page_token["updated_since"] = last_record_updated_at
+            del next_page_token["page"]
+
+        return next_page_token
+
+
+class Conversations(FreshdeskStream):
+    """Notes and Replies"""
+    def __init__(self, authenticator: AuthBase, config: Mapping[str, Any], *args, **kwargs):
+        super().__init__(authenticator=authenticator, config=config, args=args, kwargs=kwargs)
+        self.tickets_stream = Tickets(authenticator=authenticator, config=config)
```

Comment:
If you don't want to make `Conversations` stream incremental in this PR (it can be incremental too as `Tickets` is incremental), please at least enable `use_cache = True` on `Tickets`

Reply:
Yeah, done.